### PR TITLE
Refactor config

### PR DESCRIPTION
**Pull request by michael-loki**
This builds on #668 because I needed a working build 😉

### CODE REVIEW

File: llarp/net/net_addr.cpp [hunk 1]
```diff
@@ -81,25 +81,17 @@ namespace llarp
   }
 
   bool
-  Addr::from_char_array(const char* in)
+  Addr::from_char_array(string_view in)
```
Comment (michael-loki) [inline llarp/net/net_addr.cpp]: this could do with a review.

**Comment by majestrate** [comment]
this needs to be updated to reflect the fs stuff

**Comment by michael-loki** [comment]
ack

**Comment by michael-loki** [comment]
@majestrate done.

**Pull request merged**